lua/indent_blankline/init.lua

```diff
@@ -354,12 +381,25 @@ local refresh = function()
                     end
 
                     if (not indent or indent == 0) and #virtual_string == 0 then
+                        prev_indent = 0
                         vim.schedule_wrap(utils.clear_line_indent)(bufnr, i + offset)
                         return async:close()
                     end
 
+                    if not prev_indent or indent + (extra and 1 or 0) <= prev_indent + max_indent_increase then
+                        prev_indent = indent
```

Comment:
This still doesn't handle `trail_indent`
If it's off, all lines that take the indent from the previous line need to be indent one level less

```suggestion
                        prev_indent = indent - utils._if(trail_indent, 1, 0)
```

Reply:
I think this should be done in blank lines only?

Reply:
`trail_indent` seems to only affect what `get_virtual_text` returns. It doesn't affect the value of `indent`. I tried to add this and it broke.

Reply:
sorry I got the `0` and `1` switched before. Fixed it now.

With `trail_indent` off, it strips one indent level from blanklines and the extra indent.
This is how it should look like

on:
![Screen Capture_select-area_20210903195559](https://user-images.githubusercontent.com/12900252/131994781-39c247be-ee73-44d5-a76a-86362c7b6681.png)

off:
![Screen Capture_select-area_20210903195612](https://user-images.githubusercontent.com/12900252/131994805-ed5a3e64-907d-45ac-87a1-5372b103c1d8.png)

Reply:
After adding this, with `max_indent_increase = 1` only the first indent level is shown.

Reply:
Oh, I think I understood what you meant.